src/cdk-experimental/menu/menu-bar.ts

```diff
@@ -20,18 +33,88 @@ import {CDK_MENU, Menu} from './menu-interface';
   selector: '[cdkMenuBar]',
   exportAs: 'cdkMenuBar',
   host: {
+    '(keydown)': '_handleKeyEvent($event)',
+    '(focus)': 'focusFirstItem()',
     'role': 'menubar',
+    'tabindex': '0',
     '[attr.aria-orientation]': 'orientation',
   },
   providers: [
     {provide: CdkMenuGroup, useExisting: CdkMenuBar},
     {provide: CDK_MENU, useExisting: CdkMenuBar},
   ],
 })
-export class CdkMenuBar extends CdkMenuGroup implements Menu {
+export class CdkMenuBar extends CdkMenuGroup implements Menu, OnDestroy, AfterContentInit {
   /**
    * Sets the aria-orientation attribute and determines where menus will be opened.
    * Does not affect styling/layout.
    */
   @Input('cdkMenuBarOrientation') orientation: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** Handles keyboard events for the MenuBar. */
+  private _menuKeyManager?: MenuKeyManager;
+
+  /** All child MenuItem elements nested in this MenuBar. */
+  @ContentChildren(CdkMenuItem, {descendants: true})
+  private readonly _allItems: QueryList<CdkMenuItem>;
+
+  constructor(
+    private readonly _dir: Directionality,
+    @Inject(TYPE_AHEAD_DEBOUNCE) private readonly _debounceInterval: number
+  ) {
+    super();
+  }
```

Comment:
nit: format this like
```suggestion
  constructor(
      private readonly _dir: Directionality,
      @Inject(TYPE_AHEAD_DEBOUNCE) private readonly _debounceInterval: number) {
    super();
  }
```
(similar for other constructors)